### DISCUSSION TITLE
BetterDiscord is launched on the application authorization page

### DISCUSSION THREAD

**Discussion by Serega007RU** [tsukasa/BdBrowser#29] [Problems and issues]
Better Discord (or BdBrowser) starts running BetterDiscord with its plugins on the page https://discord.com/oauth2/authorize when it doesn't need to be done
You can check it on this page, on the top right Log In https://discords.com/bots/bot/621352902656524288

**Comment by tsukasa** [comment]
Greetings,

Thank you for your report. This is indeed a bug caused by BdBrowser, not BetterDiscord. `startup.js` checks Discord's UserStore for the existence of `getCurrentUser` (Function: `loadBetterDiscord`). If the UserStore is unavailable, BdBrowser registers a callback with Discord's Dispatcher to inject BetterDiscord once CONNECTION_OPEN fires.

Since the UserStore module is being loaded on the provided URL, this erroneously causes BdBrowser to assume the web client is already loaded/initialized and injects BetterDiscord directly without registering/waiting for the callback (as evident by the "getCurrentUser succeeded, running setImmediate()." log entry in the console).

I see two aspects to optimize here:
- First and foremost, the criterion for determining whether the user truly is in the web client should validate at least one more additional condition. Querying both the GuildStore and the UserStore seemingly does the trick and is my current hotfix.

- Even with the fix described above, BdBrowser still does open the VFS and registers the callback (although it will never be called). The `preload.js` getting injected also is not ideal, despite it not having any negative impact. The front-end initialization process should exit early if the user is not in the web client.

To address the immediate issue, I have [pushed a fix](https://github.com/tsukasa/BdBrowser/releases/tag/v1.9.5.20231111) containing the changes detailed in the first aspect. The refactoring work to address the underlying problem holistically will probably take some time and will be part of a future release.
